src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -256,6 +260,47 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
         }
     }
 
+    private loadController(notebookDocument: NotebookDocument) {
+        // Immediately try to find a selected controller for our NotebookDocument,
+        // as it's possible that a selection event fired before our ctor was able to run
+        const controller = this.notebookControllerManager.getSelectedNotebookController(notebookDocument);
+        if (controller !== undefined) {
+            this.registerKernel(notebookDocument, controller);
+            this.initialControllerSelected.resolve();
+        }
+
+        // Ensure we hear about any controller changes so we can update our cache accordingly
+        this.notebookControllerManager.onNotebookControllerSelected(
```

Comment:
I thought you didn't need to register the kernel anymore. You just need the notebook document? Then you always ask for the active kernel for that document.

Reply:
Yup, I'm deferring kernel changes to a separate PR (want to keep this one focused on the ctor refactor)